group-coordinator/src/main/java/org/apache/kafka/coordinator/group/classic/ClassicGroupMember.java

```diff
@@ -199,6 +199,17 @@ public byte[] metadata(String protocolName) {
             protocolName);
     }
 
+    /**
+     * Get the metadata of any supported protocol.
+     */
+    public byte[] metadata() {
+        for (JoinGroupRequestProtocol supportedProtocol : supportedProtocols) {
+            return supportedProtocol.metadata();
+        }
+
+        throw new IllegalArgumentException("Member does not support any protocol.");
```

Comment:
Interesting way to code this. Intuitively, I would have check if supportedProtocols.isEmpty() and throw if it is and then get the first element. I find it a bit more explicit.